grpc/interceptors.go

```diff
@@ -205,6 +214,15 @@ func (smi *serverMetadataInterceptor) observeLatency(clientReqTime string) error
 	// Calculate the elapsed time since the client sent the RPC
 	reqTime := time.Unix(0, reqTimeUnixNanos)
 	elapsed := smi.clk.Since(reqTime)
+
+	// If the elapsed time is very large, that indicates it is probably due to
+	// clock skew rather than simple latency. Refuse to handle the request, since
+	// accurate timekeeping is critical to CA operations and large skew indicates
+	// something has gone very wrong.
+	if tooSkewed(elapsed) {
+		return fmt.Errorf("dangerously large gRPC clock skew: %s", elapsed)
```

Comment:
```suggestion
		return fmt.Errorf("gRPC client reported a very different time: %s (client) vs %s (this server)",
		    reqTime, smi.clk.Now())
```

This is likely to be more helpful in debugging what is wrong.